ducanhpham0312/zeevision
# feat: audit log table and bpmn coloring

### Linked issue

Closes ducanhpham0312/zeevision-private#150


### Description

- Audit Log table to SingleInstancePage
- Add coloring to BPMN diagram depends on the audit log state
- Fix a bug where resizing a window including bpmn diagram breaks the application
- Update sample query to only include necessary auditlog queries.
- "Incompatible" incident queries are also modified.

### Testing

- Run test bench, start the application and run the query in pgAdmin.
- Open `/instances/11` and verify that some elements in the BPMN diagram are colored (green border, red-filled, green-filled). 
  - Green border: The element has completed or have been resolved from an incident
  - Green-filled: The element is in the activating - completing phase (requires user focus)
  - Red-filled: There is an incident occur in the element (requires user focus)
  - Gray-filled: The element is terminated.
  - The arrow is always green (I cannot color the triangle though :/ ).
  - Please also comment and make suggestion on color and opacity selection etc.
- To test the color change when an incident occurs:
  - Deploy the money-loan process
  - Create an instance without any variable. The gateway in the bpmn should be red (where incident occurs)
  - Open ZSM with that instance and set the variable (e.g., debt : 100). Then switch to the incident tab and hit resolve. The gateway should now have green border color.
- The `Audit Log` table are visible after choosing the corresponding tab.
- Open SingleProcessPage or SingleInstancePage and try resizing the browser window. It should not break the app.
